util/CosmosHelper.js

```diff
@@ -48,94 +51,92 @@ export function amountToLIKE(likecoin) {
-    isFailed: (code && code !== '0') || !fullLogs[0].success,
+    isFailed: (code && code !== '0'),
   };
 }
 
 export async function getTransferInfo(txHash, opt) {
-  if (!cosmosClient) await initCosmosClient();
+  if (!stargateClient) await initStargateClient();
   const { blocking } = opt;
-  // TODO: handle tranferMultiple?
-  let txData = await cosmosClient.getTx(txHash);
+  let txData = await stargateClient.getTx(txHash);
   if ((!txData || !txData.height) && !blocking) {
     return {};
   }
   while ((!txData || !txData.height) && blocking) {
     await timeout(1000); // eslint-disable-line no-await-in-loop
-    txData = await cosmosClient.getTx(txHash); // eslint-disable-line no-await-in-loop
+    txData = await stargateClient.getTx(txHash); // eslint-disable-line no-await-in-loop
   }
   if (!txData) throw new Error('Cannot find transaction');
   const {
-    rawLog,
-    timestamp,
+    height,
     code,
-    tx: {
-      value: {
-        msg,
-      },
-    },
   } = txData;
-  let amount = [];
-  let from = [];
-  let to = [];
-  if (msg.length > 1) {
-    msg.forEach((m) => {
-      const {
-        value: {
-          amount: [a],
-          from_address: f,
-          to_address: t,
-        },
-      } = m;
-      amount.push(a);
-      from.push(f);
-      to.push(t);
-    });
-  } else {
-    ([{
-      value: {
+  const block = await stargateClient.getBlock(height);
+  const { header: { time: timestamp } } = block;
+  const txRaw = TxRaw.decode(Buffer.from(txData.tx, 'base64'));
+  const txBody = TxBody.decode(txRaw.bodyBytes);
+  const { messages } = txBody;
+  let amounts;
+  let amount;
+  let from;
+  let to;
+  const { typeUrl, value } = messages[0];
+  if (typeUrl === '/cosmos.bank.v1beta1.MsgSend' || typeUrl === '/cosmos.bank.v1beta1.MsgMultiSend') {
+    if (typeUrl === '/cosmos.bank.v1beta1.MsgSend') {
+      const payloadValue = MsgSend.decode(value);
+      ({
         amount: [amount],
-        from_address: from,
-        to_address: to,
-      },
-    }] = msg);
+        fromAddress: from,
+        toAddress: to,
+      } = payloadValue);
+      amounts = amount;
+    } else if (typeUrl === '/cosmos.bank.v1beta1.MsgMultiSend') {
+      const payloadValue = MsgMultiSend.decode(value);
+      const {
+        inputs,
+        outputs,
+      } = payloadValue;
+      from = inputs.length > 1 ? inputs.map(i => i.address) : inputs[0].address;
+      to = outputs.length > 1 ? outputs.map(o => o.address) : outputs[0].address;
+      amounts = outputs.length > 1 ? outputs.map(o => o.coins[0]) : [outputs[0].coins[0]];
+    }
   }
+
   if (Array.isArray(from)) [from] = from; // TODO: support multiple from addr?
   if (!txData.height) {
     return {
       _from: from,
       _to: to,
-      _amount: amount,
+      _amount: amounts,
     };
   }
-  const fullLogs = JSON.parse(rawLog.slice(1, rawLog.length - 1));
-  const isFailed = (code && code !== '0') || !fullLogs.success;
+  const isFailed = (code && code !== '0');
```

Comment:
Is `code` a string or a number in the new cosmjs getTx() result?

Reply:
it's now a number. Will change it to !== 0